### ISSUE TITLE
Deleting a backing server bucket causes sync_gateway internal error

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/sync_gateway#1737]
This appears to be a regression

sync_gateway version: Couchbase Sync Gateway/1.3.0(143;062bc26)
Couchbase Server version: 4.1.1-5914

Scenario:
A sync_gateway db has a backing Couchbase server bucket
A bunch of the sg REST endpoints are hit
The backing server bucket is deleted
sg REST endpoints are hit again to validate that endpoint that should still be reachable are

Expected: 
`GET http://s61406cnt72.sc.couchbase.com:4984/db/` should still be reachable with a state of "offline"
Actual:
`GET http://s61406cnt72.sc.couchbase.com:4984/db/` returns 

```
{
"error": "Internal Server Error",
"reason": "Internal error: EOF"
}
```

**Comment by ajres** [comment]
On a local test with SG master and CBS 4.5.0 I get a different error:

```
{"error":"Bad Gateway","reason":"Auth failure (0x20)"}
```

I only get an error for the .../db/ end-point, for specific resource end-points, e.g.

`.../db/doc1` or `.../db/_changes`

I get the expected `service unavailable` error message:

`{"error":"Service Unavailable","reason":"DB is currently under maintenance"}`

**Comment by ajres** [comment]
This was fixed on 1.2.1, looks like it was not merged back to master.
